### PR TITLE
*:Fix the bug of consistent plan but inconsistent plan digest

### PR DESCRIPTION
**Pull request by jyz0309**
Signed-off-by: jyz0309 <45495947@qq.com>

### What problem does this PR solve?

Problem Summary:
When the execution plan is the same, the number of plans displayed in tidb-dashboard may be inconsistent and more because of the partition table.
For example:
```
# Plan_digest: 37e378a2cb4384941179b1bb16525596bcde0775242e136aabceb385a437d6bc
select * from t3 where a<5;
```
```
# Plan_digest: 47507992976b9903d5d2e85b2a1973c2f249769e387576dc19ee8500fb8fd6d9
select * from t3 where a<20;
```
### What is changed and how it works?

What's Changed:
Move the Partition field from PointGetPlan AccessObject() function to OberatorInfo function. 

How it Works:
When tidb need to get the PointGetPlan digest, we can make `partition:?` by setting `normalized` true.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Manual test (add detailed scripts or steps below)

Side effects

- Performance regression
    - Consumes more CPU

### Release note 
- Fix the bug of consistent PointGetPlan but inconsistent plan digest

### CODE REVIEW

**Review by eurekaka — COMMENTED** [review]
Please add test for this PR and provide a release note in the PR description.

File: planner/core/explain.go [hunk 1]
```diff
@@ -64,27 +64,30 @@ func (p *PhysicalIndexScan) ExplainID() fmt.Stringer {
...
 	fmt.Fprintf(buffer, "table:%s", tblName)
 	if p.isPartition {
-		if pi := p.Table.GetPartitionInfo(); pi != nil {
+		if normalized {
+			fmt.Fprintf(buffer, ", partition:?")
+		} else if pi := p.Table.GetPartitionInfo(); pi != nil {
 			partitionName := pi.GetNameByID(p.physicalTableID)
 			fmt.Fprintf(buffer, ", partition:%s", partitionName)
 		}
+
```
Comment (crazycs520) [inline planner/core/explain.go:90]: remove this blank line

Comment (jyz0309) [reply planner/core/explain.go]: done

**Comment by jyz0309** [comment]
> Please add test for this PR and provide a release note in the PR description.

done, PTAL

File: planner/core/testdata/plan_normalized_suite_out.json [hunk 2]
```diff
@@ -197,6 +197,20 @@
...
         ]
+      },
+      {
+        "SQL": "insert into t3 values (4,20),(30,30)",
+        "Plan": [
+          ""
+        ]
+      },
+      {
+        "SQL": "select * from t3 where a<5",
```
Comment (crazycs520) [inline planner/core/testdata/plan_normalized_suite_out.json:208]: Need add test for partition table?

File: planner/core/explain.go [hunk 3]
```diff
@@ -64,24 +64,26 @@ func (p *PhysicalIndexScan) ExplainID() fmt.Stringer {
...
 	buffer := bytes.NewBufferString("")
 	tblName := p.Table.Name.O
 	if p.TableAsName != nil && p.TableAsName.O != "" {
 		tblName = p.TableAsName.O
 	}
 	fmt.Fprintf(buffer, "table:%s", tblName)
 	if p.isPartition {
-		if pi := p.Table.GetPartitionInfo(); pi != nil {
+		if normalized {
+			fmt.Fprintf(buffer, ", partition:?")
```
Comment (crazycs520) [inline planner/core/explain.go:85]: Which test will cover this?

**Review by crazycs520 — COMMENTED** [review]
LGTM

**Comment by crazycs520** [comment]
@tiancaiamao PTAL

**Comment by tiancaiamao** [comment]
LGTM

**Comment by tiancaiamao** [comment]
PTAL @eurekaka

**Review by eurekaka — APPROVED** [review]
LGTM

**Pull request merged**